### ISSUE TITLE
Definition of wing spar caps

### ISSUE THREAD

**Issue by hardt-fa-flb** [DLR-LY/CPACS#558]
The current definition of the spar caps only consists of an area and a material. This is not accurate enough to model properly a cross section e.g. for sailplanes or wind turbine blades.

![grafik](https://user-images.githubusercontent.com/41366119/42872930-77eb9186-8a7e-11e8-9ceb-bd3d2e7ae3bd.png)

FA-FLB recommends a new definition of the wing spar caps:

![grafik](https://user-images.githubusercontent.com/41366119/42872932-7b553eee-8a7e-11e8-80da-4522811932ed.png)
 
Spar Caps material is added to the inner side of the wing shell.
For the definition of one spar cap (upper or lower) the start and end point on the wing shell has to be defined, so two positions are needed. The positions can be defined in two ways: First as a Xsi coordinate of the actual wing cross section and second relative to the intersection point from the web with the wing shell. From this intersection point, a Xsi difference is given for the point definition.
For simplification and less redundancy, we recommend to remove the web2 element. If a spar with two webs is required, you can use two spars with one web each.

Requested changes to CPACS:

- sparCrossSection Element:
    - Remove of web2
    - Rename web1 to web
- capType Element:
    - Remove of area
    - Add position1 and position2 elements (Type: SparCapPosition Element)
- SparCapPosition Element:
    - Selection between definition in wing Xsi coordinates or Delta Xsi as distance from the intersection point of the web with the wing shell:
         - wingXsi or
         - deltaXsi

**Comment by MarAlder** [comment]
I agree with your request for a more detailed caps definition. Some questions from my side:

- Maybe we should not remove `area`. Would a choice element between `area` and a more detailed desciption be a ok?

- With `wingXsi` the length of the cap depends on the profile thickness. Is this needed? What if the spar-wing intersection is outside of the cap (e.g., due to rotation).

- Furthermore, since the capType is also used for ribs, it is not always the xsi coordinate (see #628); Can we define a more general solution here?

**Comment by sdeinert** [comment]
Would it be an option for you to specify the cap geometry through
- Width1 (forward of the spar web)
- Width2 (rear of the spar web)
- Thickness (height from loft inner side to inner side of cap)

With this information the area of the cap could be calculated directly with less geometric construction effort.

This could be added as a choice alternative to directly providing area allowing both options depending on the level of detail at the current development state.

**Comment by hardt-fa-flb** [comment]
@MarAlder:
- A choice between the current definition and a more detailed definition is fine.
- The dependence of the cap width on the profile thickness (or rather the chord length?) is not needed. Maybe the width can be defined absolute rather than relative? I would also suggest that the cap width in front of and behind the spar is measured from the intersection of the web and wing shell and not from the spar position on the chord line.
- I didn't know that the capType is also used for the ribs. what would you suggest there?

@sdeinter: The definition of the cap with absolute length instead of relative length is fine. However, the thickness of the cap in our application results from the corresponding composite material and would be redundant at this point. If the area is still needed, I think a choice between a simplified and a more detailed cap definition makes more sense.
